security/src/main/java/io/micronaut/security/authentication/Authentication.java

```diff
@@ -40,4 +41,18 @@
      */
     @NonNull
     Map<String, Object> getAttributes();
+
+    /**
+     * Retrieves an explicitly typed attribute value.
+     *
+     * @param name the name of the attribute
+     * @param clazz the expected class of the attribute
+     * @param <T> the expected type of the attribute
+     * @return the typed attribute or null if it doesn't exist or the type is not correct
+     */
+    default <T> Optional<T> getAttribute(String name, @NonNull Class<T> clazz) {
```

Comment:
Could you annotate the return type with `@NonNull`

Could you annotate the parameter `name` with `@NonNull`